src/main/scala/org/allenai/deep_qa/data/BackgroundCorpusSearcher.scala

```diff
@@ -126,3 +146,33 @@ class LuceneBackgroundCorpusSearcher(params: JValue) extends BackgroundCorpusSea
     return true
   }
 }
+
+
+// trait used for handling various types of queries
+trait Query {}
+
+
+// Query type for basic string queries
+class StringQuery(var query: String = "") extends Query {
+
+  def set(queryIn: String): Unit = {
+    query = queryIn
+  }
+}
+
+
+// Query type for boosting different portions of the query
+class BoostedQuery(boostedIn: Seq[(String, Float)]) extends Query () {
```

Comment:
Again here, I don't think you actually need mutability.  You can just do `case class BoostedQuery(boostedQueries: Seq[(String, Float)]) extends Query`.